app/src/main/java/com/yml/womensafety/navigationdrawer/home/HomePageActivity.kt

```diff
@@ -52,5 +89,125 @@ class HomePageActivity : AppCompatActivity() {
+            Log.e(LOG_MESSAGE, exception.toString())
+        }
+        val checkInternetPermission: String = Manifest.permission.INTERNET
+        try {
+            if (ActivityCompat.checkSelfPermission(
+                    this,
+                    checkInternetPermission
+                ) != PackageManager.PERMISSION_GRANTED
+            ) {
+                ActivityCompat.requestPermissions(
+                    this,
+                    arrayOf(checkPermission),
+                    Request_Code_permission
+                )
+            }
+        } catch (exception: Exception) {
+            Log.e(LOG_MESSAGE, exception.toString())
+        }
+    }
+
+    //This method will send the SMS along with location
+    private fun sendSms() {
+        contactsViewModel.getContactsList(object : ContactsResponse {
+            override fun onContactsReceiveSuccess(userContacts: List<String>) {
+                try {
+                    val gpsTracker = GPSTracker(applicationContext)
+                    val location: Location? = gpsTracker.getLocation()
+                    if (location != null) {
+                        val locationLatitude: Double = location.latitude
+                        val locationLongitude: Double = location.longitude
+                        var strAdd = ""
+                        val geoCoder = Geocoder(this@HomePageActivity, Locale.getDefault())
+                        try {
+                            val addresses =
+                                geoCoder.getFromLocation(locationLatitude, locationLongitude, 1)
+                            if (addresses != null) {
+                                val returnedAddress = addresses[0]
+                                val strReturnedAddress = StringBuilder("")
+                                for (i in 0..returnedAddress.maxAddressLineIndex) {
+                                    strReturnedAddress.append(returnedAddress.getAddressLine(i))
+                                        .append("\n")
+                                }
+                                strAdd = strReturnedAddress.toString()
+                            }
+                        } catch (e: Exception) {
+                            e.printStackTrace()
+                        }
+                        val message =
+                            getString(R.string.location) + "$locationLatitude,$locationLongitude\n$strAdd"
+                        val smsManager = SmsManager.getDefault()
+                        userContacts.forEach {
+                            smsManager.sendTextMessage(it, null, message, null, null)
+                        }
+                        Toast.makeText(
+                            applicationContext, getString(R.string.message_sent),
+                            Toast.LENGTH_LONG
+                        ).show()
+                    }
+
+                } catch (ex: ActivityNotFoundException) {
+                    Toast.makeText(
```

Comment:
Same here?

Reply:
If the message is not sent this will be displayed.